src/source/PeerConnection/SessionDescription.c

```diff
@@ -666,13 +666,15 @@ STATUS copyTransceiverWithCodec(PKvsPeerConnection pKvsPeerConnection, RTC_CODEC
     *pDidFindCodec = FALSE;
 
     CHK_STATUS(doubleListGetHeadNode(pKvsPeerConnection->pTransceievers, &pCurNode));
-    while (pCurNode != NULL && pTargetKvsRtpTransceiver == NULL) {
+    while (pCurNode != NULL) {
         CHK_STATUS(doubleListGetNodeData(pCurNode, &data));
-        pCurNode = pCurNode->pNext;
         pKvsRtpTransceiver = (PKvsRtpTransceiver) data;
         if (pKvsRtpTransceiver != NULL && pKvsRtpTransceiver->sender.track.codec == rtcCodec) {
             pTargetKvsRtpTransceiver = pKvsRtpTransceiver;
+            doubleListDeleteNode(pKvsPeerConnection->pTransceievers, pCurNode);
```

Comment:
This will not delete the underlying object so you will leak memory.

Reply:
the purpose of this method is to sort transceivers, so i delete it here and add to the end of linked list on line 680